core/components/batchedWriterAssembly.go

```diff
@@ -0,0 +1,112 @@
+// Copyright 2015-2017 trivago GmbH
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package components
+
+import (
+	"github.com/sirupsen/logrus"
+	"github.com/trivago/gollum/core"
+	"io"
+	"time"
+)
+
+// BatchedWriterAssembly is a helper struct for io.Writer compatible classes that use batch directly for resources
+type BatchedWriterAssembly struct {
+	Batch           core.MessageBatch // Batch contains the MessageBatch
+	Created         time.Time         // Created contains the creation time from the writer was set
+	writer          BatchedWriter
+	assembly        core.WriterAssembly
+	flushTimeout    time.Duration // max sec to wait before a flush is aborted
+	batchTimeout    time.Duration // max sec to wait before batch will flushed
+	batchFlushCount int
+	logger          logrus.FieldLogger
+}
+
+// BatchedWriter is an interface for different file writer like disk, s3, etc.
+// TODO: Replace methods with os.FileInfo interface
```

Comment:
I think this interface is fine as it is.
os.FileInfo introduces functions not required by the functions using BatchedWriter.
Function that are not used should not be part if the interface.